lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Checkbox,
+  CheckboxCheckedFilledIcon,
+  Input,
+  RadioGroup,
+} from '@hypothesis/frontend-shared';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useId } from 'preact/hooks';
+
+export type GradingType = 'all_or_nothing' | 'scaled';
+
+export type AutoGradingConfig = {
+  /** Whether auto grading is active for the assignment or not */
+  active?: boolean;
+
+  /**
+   * - all_or_nothing: students need to meet a minimum value, making them get
+   *                   either 0% or 100%
+   * - scaled: students may get a proportional grade based on the amount of
+   *           annotations. If requirement is 4, and they created 3, they'll
+   *           get a 75%
+   */
+  gradingType: GradingType;
+
+  /**
+   * - cumulative: both annotations and replies will be counted together for
```

Comment:
"cumulative" doesn't seem quite the right choice of word here. "Combined" would seem the more obvious choice to me since it is the opposite of "separate". The word "cumulative" does appear in the UI design though, so we'd need to discuss changes there first.

Reply:
We are going to go with cumulative for now https://hypothes-is.slack.com/archives/C2C2U40LW/p1725368924377989?thread_ts=1725366320.845439&cid=C2C2U40LW

We could decide to use a different term internally though, but I think that would be confusing.

That could happen anyway though, if we ship it as is and start to "store" the `cumulative` value, and then we decide to change it because users find it confusing.